spec/lib/graphql/resource_spec.rb

```diff
@@ -566,6 +573,231 @@ def build_resource(stubbed_class)
+        results = schema.execute(create_mutation, variables: { attributes: { name: name, organizationId: organization.id } }, context: { current_user: current_user })
+        data = results["data"]
+
+        aggregate_failures do
+          expect(data["createAdvisor"]["resource"]).to be_nil
+          expect(data["createAdvisor"]["errors"]["resource"]).to eql "Unauthorized action for Advisor create"
+          expect(Advisor.where(name: name).exists?).to eql false
+        end
+      end
+
+      it "creates with restricted arguments" do
+        organization = FactoryBot.create(:organization)
+        role = FactoryBot.build(:role, organization: organization)
+        current_user = FactoryBot.create(:user, role: role)
+        advisorBaseResource = FactoryBot.build(:resource,
+          name: "Advisor",
+          alias: "Advisor",
+          resource_type_id: "HasHelpers::ResourceType::::BaseResource",
+          parent: nil,
+          field_resource: nil
+        )
+        orgBaseResource = FactoryBot.build(:resource,
+          name: "Organization", resource_type_id: "HasHelpers::ResourceType::::BaseResource",
+          alias: "Organization",
+          parent: nil,
+          field_resource: nil
+        )
+        associationResource = FactoryBot.build(:resource,
+          name: "optional_org_id",
+          alias: "optional_org_id",
+          field_resource: orgBaseResource,
+          field_class_name: "Organization",
+          resource_type_id: "HasHelpers::ResourceType::::Field",
+          parent: advisorBaseResource)
+        fieldResource = FactoryBot.build(:resource,
+          name: "nickname",
+          alias: "nickname",
+          resource_type_id: "HasHelpers::ResourceType::::Field",
+          parent: advisorBaseResource,
+          field_resource: nil
+        )
+        FactoryBot.create(:restriction,
+          resource: fieldResource,
+          restriction_operation_id: "HasHelpers::RestrictionOperation::::Create",
+          role: role,
+          organization: organization
+        )
+        FactoryBot.create(:restriction,
+          resource: associationResource,
+          restriction_operation_id: "HasHelpers::RestrictionOperation::::Create",
+          role: role,
+          organization: organization
+        )
+
+        name = "Bob"
+        nickname = "Michael"
+        results = schema.execute(create_mutation,
+          variables: {
+            attributes: { name: name, nickname: nickname, organizationId: organization.id, optionalOrgId: organization.id }
+          },
+          context: { current_user: current_user }
+        )
+        data = results["data"]
+
+        aggregate_failures do
+          expect(data["createAdvisor"]["resource"]["nickname"]).to be_nil
+          expect(data["createAdvisor"]["resource"]["optionalOrg"]).to be_nil
+          expect(data["createAdvisor"]["errors"]["warning"]["advisor"].size).to eql 2
+          expect(Advisor.where(name: name).exists?).to eql true
```

Comment:
why was the advisor created if the user had restrictions on those fields? Or was the advisor created but the fields that were restricted just were filled with nil ?

Reply:
the second description is the correct one, advisor will be created but with restricted attributes filtered. User will receive a warning about the attributes that were not saved because of role's restrictions